lib/spack/spack/fetch_strategy.py

```diff
@@ -647,6 +654,8 @@ def __init__(self, **kwargs):
 
         self._git = None
         self.submodules = kwargs.get('submodules', False)
+        self.full_depth = kwargs.get('full_depth', False)
+        self.all_branches = kwargs.get('all_branches', False)
```

Comment:
Is it worth adding checks and at least logging warnings if there is an inconsistency between this option and the git version (e.g., version < 1.7.10 and not `all_branches`).

Reply:
Possibly, but this would be adding a warning for behavior which is currently entirely normal. In other words, we already have the situation where the behavior is to use `--single-branch` or `--depth 1` if the current version of git supports them, and silently not use them if not. Happy to add the warnings if you want them, but I worry about the value of warnings that could be quite frequent on older systems, while still not requiring action on the part of the user. The only consequence of _not_ using `--depth 1` or `--single-branch` is a bigger local checkout of the git repository than otherwise.

Reply:
So it sounds like it isn't necessarily useful.  Thanks for the explanation.